components/docker-auth/pkg/auth/authn_handler.go

```diff
@@ -207,112 +213,108 @@ func validateAccessToken(token string, providedUsername string, execId string) b
 	client := &http.Client{Transport: tr}
 	res, err := client.Do(req)
 	if err != nil {
-		log.Printf("[%s] Error sending the request to the introspection endpoint : %s\n", execId, err)
-		return false
+		return false, fmt.Errorf("error sending the request to the introspection endpoint : %s", err)
 	}
 
 	if res.StatusCode == http.StatusBadRequest {
-		log.Printf("[%s] %d status code returned from IDP probably due to empty token\n", execId,
+		return false, fmt.Errorf("[%s] %d status code returned from IDP probably due to empty token", execId,
 			res.StatusCode)
-		return false
 	} else if res.StatusCode != http.StatusOK {
-		log.Printf("[%s] Error while calling IDP, status code :%d. Exiting without authorization\n", execId,
-			res.StatusCode)
-		return false
+		return false, fmt.Errorf("[%s] Error while calling IDP, status code :%d. Exiting without "+
+			"authorization\n", execId, res.StatusCode)
 	}
 
 	defer res.Body.Close()
 	body, err := ioutil.ReadAll(res.Body)
 	if err != nil {
-		log.Printf("[%s] Error reading the response from introspection endpoint. Returing without "+
-			"authorization : %s\n", execId, err)
-		return false
+		return false, fmt.Errorf("error reading the response from introspection endpoint. Returing without "+
+			"authorization : %s", err)
 	} else {
-		log.Printf("[%s] Response recieved from introspection endpoint : %s\n", execId, body)
+		logger.Debugf("[%s] Response recieved from introspection endpoint : %s", execId, body)
 	}
 
 	var result map[string]interface{}
 	err = json.Unmarshal([]byte(string(body)), &result)
 	if err != nil {
-		log.Printf("[%s] Error un marshalling the json : %s\n", execId, err)
-		return false
+		return false, fmt.Errorf("error un marshalling the json : %s", err)
 	}
 	isActive, ok := (result["active"]).(bool)
 	if !ok {
-		log.Printf("[%s] Error casting active to boolean. This may be due to a invalid token\n", execId)
-		return false
+		return false, fmt.Errorf("error casting active to boolean. This may be due to a invalid token")
+	}
+	logger.Debugf("[%s] Resolved acess token validity", execId)
+	isExpired, err := isExpired(result["exp"], logger, execId)
```

Comment:
Is there any reason to use map[string]interface{} without using struct?

Reply:
Fixed